Modules/Build.lua

```diff
@@ -828,6 +839,12 @@ function buildMode:OnFrame(inputEvents)
 	-- Update contents of main skill dropdowns
 	self:RefreshSkillSelectControls(self.controls, self.mainSocketGroup, "")
 
+	-- Delete any possible fabricated groups
+	if checkFabricatedGroups then
+		deleteFabricatedGroup(self.skillsTab)
+		checkFabricatedGroups = false
+	end
+
```

Comment:
No, it can't. It needs to be where it is and hence it was implemented this way. 

```
	-- Update contents of main skill dropdowns
	self:RefreshSkillSelectControls(self.controls, self.mainSocketGroup, "")
```
Will rebuild it and never make it go away if you move it into the "if self.buildFlag then" block higher up.

Reply:
Ah, fair enough, I was just looking at the checkFabricatedGroups flag as being redundant.

Reply:
the `self.buildFlag` is reset to `false` when it's handled in it's conditional if statement above, hence I needed a variable to make a copy of it so that after `self:RefreshSkillSelectControl()` is done I know I still need to remove a skill group.